### PR TITLE
chore: update persistent-merkle-tree 0.6.1

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

- Migrate to latest persistent-merkle-tree
- See https://github.com/ChainSafe/lodestar/pull/5959#issuecomment-1723527505

**Description**

- Use `setHasher()` at the beginning to enforce `as-sha256` in lodestar

### CODE REVIEW

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -1,11 +1,18 @@
 #!/usr/bin/env node
 
-// MUST import first to apply preset from args
-import "./applyPreset.js";
-import {YargsError} from "./util/index.js";
-import {getLodestarCli, yarg} from "./cli.js";
+// eslint-disable-next-line no-restricted-imports, import/no-extraneous-dependencies
+import {hasher} from "@chainsafe/persistent-merkle-tree/lib/hasher/as-sha256.js";
+// eslint-disable-next-line no-restricted-imports, import/no-extraneous-dependencies
+import {setHasher} from "@chainsafe/persistent-merkle-tree/lib/hasher/index.js";
```
Comment (wemeetagain) [inline packages/cli/src/index.ts]: put all the changes here the applyPreset file

**Comment by twoeths** [comment]
Screenshot 2023-10-11 at 17 48 15

I deployed to feat1 mainnet node, it does not work as expected (stable mainnet node takes < 58ms consistently). Could that be an issue of `setHasher()` itself @wemeetagain ?

**Comment by twoeths** [comment]
within lodestar, I think there are 2 persistent-merkle-tree:

```
find node_modules/ -type d -name "persistent-merkle-tree"
node_modules/@chainsafe/persistent-merkle-tree
node_modules/@chainsafe/ssz/node_modules/@chainsafe/persistent-merkle-tree
```

we may need to set our hasher for both of them

**Comment by twoeths** [comment]
this is due to `node_modules/@chainsafe/ssz/node_modules/@chainsafe/persistent-merkle-tree` being cached in the deployment. This branch works fine after removing `node_modules`

Screenshot 2023-10-12 at 16 58 37

**Review by wemeetagain — DISMISSED** [review]

**Review by wemeetagain — APPROVED** [review]

File: packages/cli/src/applyPreset.ts [hunk 2]
```diff
@@ -1,4 +1,13 @@
 // MUST import this file first before anything and not import any Lodestar code.
+
+// eslint-disable-next-line no-restricted-imports, import/no-extraneous-dependencies
+import {hasher} from "@chainsafe/persistent-merkle-tree/lib/hasher/as-sha256.js";
+// eslint-disable-next-line no-restricted-imports, import/no-extraneous-dependencies
+import {setHasher} from "@chainsafe/persistent-merkle-tree/lib/hasher/index.js";
+
+// without setting this first, persistent-merkle-tree will use noble instead
+setHasher(hasher);
```
Comment (nflaig) [inline packages/cli/src/applyPreset.ts:9]: Does it make sense to set `as-sha256` as default in the library? Consumers still have the option to use noble but it is then their responsibility and we don't have to make sure it is set correctly in Lodestar.

Comment (twoeths) [reply packages/cli/src/applyPreset.ts:9]: it's decided to use noble as default implementation, see https://github.com/ChainSafe/ssz/issues/313

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.12.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.12.0) :tada:

**Pull request merged**